packages/cli/src/generator.ts

```diff
@@ -304,3 +304,49 @@ export function getImportMetaUrl() {
     factory.createIdentifier('url'),
   );
 }
+
+/**
+ * Check if the TypeScript version supports import attributes.
+ *
+ * @returns `true` if the TypeScript version supports import attributes, or
+ * `false` otherwise.
+ */
+export function hasImportAttributes() {
+  return 'createImportAttributes' in factory;
+}
+
+/**
+ * Get the import attributes for the given name and value.
+ *
+ * This function supports older versions of TypeScript by using import
+ * assertions rather than import attributes, if necessary. Import assertions are
+ * deprecated though, so it is recommended to use a recent version of TypeScript
+ * that supports import attributes.
+ *
+ * @param name - The name of the attribute.
+ * @param value - The value of the attribute.
+ * @returns The import attributes.
+ */
+export function getImportAttribute(name: string, value: string) {
+  /* istanbul ignore next if -- @preserve */
+  if (hasImportAttributes()) {
+    return factory.createImportAttributes(
+      factory.createNodeArray([
+        factory.createImportAttribute(
+          factory.createIdentifier(name),
+          factory.createStringLiteral(value),
+        ),
+      ]),
+    );
+  }
+
+  /* istanbul ignore next -- @preserve */
```

Comment:
Can we not test this part?

Reply:
The `factory.createAssertClause` call. It's only used for TypeScript <5.3, and not easy to test without mocking something. We could potentially mock `hasImportAttributes()` instead of ignoring this though.